src/Core/BaseKernel.php

```diff
@@ -264,6 +264,7 @@ protected function bootErrorHandling()
      */
     public function getEnvironment()
     {
+        Deprecation::notice('4.12.0', 'Use Director::get_environment_type() instead. Since 5.0 it should return only if kernel overrides. No checking SESSION or Environment.');
```

Comment:
```suggestion
        Deprecation::notice('4.12.0', 'Use Director::get_environment_type() instead.');
```
No need for all that extra info - we want these messages to give devs the information they immediately need to upgrade. "Use x instead" on its own is sufficient and is consistent with other similar deprecations.
Also update the version and message in the phpdoc above.

Reply:
Updated